python/huggingfaceserver/huggingfaceserver/vllm_completions.py

```diff
@@ -0,0 +1,452 @@
+                    yield response_json
+        except ValueError as e:
+            data = self.create_streaming_error_response(str(e))
+            yield data
+
+    def request_output_to_completion_response(
+        self,
+        final_res_batch: List[RequestOutput],
+        request: CreateCompletionRequest,
+        request_id: str,
+        created_time: int,
+        model_name: str,
+    ) -> Completion:
+        choices = []
+        num_prompt_tokens = 0
+        num_generated_tokens = 0
+        for final_res in final_res_batch:
+            assert final_res is not None
+            prompt_token_ids = final_res.prompt_token_ids
+            prompt_logprobs = final_res.prompt_logprobs
+            prompt_text = final_res.prompt
+
+            for output in final_res.outputs:
+                if request.echo and request.max_tokens == 0:
+                    token_ids = prompt_token_ids
+                    top_logprobs = prompt_logprobs
+                    output_text = prompt_text
+                elif request.echo and request.max_tokens > 0:
+                    token_ids = prompt_token_ids + output.token_ids
+                    top_logprobs = prompt_logprobs + output.logprobs
+                    output_text = prompt_text + output.text
+                else:
+                    token_ids = output.token_ids
+                    top_logprobs = output.logprobs
+                    output_text = output.text
+
+                if request.logprobs is not None:
+                    logprobs = self._create_logprobs(
+                        token_ids=token_ids,
+                        top_logprobs=top_logprobs,
+                        num_output_top_logprobs=request.logprobs,
+                    )
+                else:
+                    logprobs = None
+
+                choice_data = CompletionChoice(
+                    index=len(choices),
+                    text=output_text,
+                    logprobs=logprobs,
+                    finish_reason=output.finish_reason,
+                    stop_reason=output.stop_reason,
+                )
+                choices.append(choice_data)
+
+            num_prompt_tokens += len(prompt_token_ids)
+            num_generated_tokens += sum(
+                len(output.token_ids) for output in final_res.outputs
+            )
+
+        usage = CompletionUsage(
+            prompt_tokens=num_prompt_tokens,
+            completion_tokens=num_generated_tokens,
+            total_tokens=num_prompt_tokens + num_generated_tokens,
+        )
+
+        return Completion(
+            id=request_id,
+            created=created_time,
+            model=model_name,
+            object="text_completion",
+            choices=choices,
+            usage=usage,
+        )
+
+    def apply_chat_template(self, messages: Iterable[ChatCompletionMessageParam]):
+        return self.tokenizer.apply_chat_template(conversation=messages, tokenize=False)
+
+    def create_error_response(
```

Comment:
We probably do not need this as kserve has error handles registered already
https://github.com/kserve/kserve/blob/9fd262f408ce5939d3d6e179fa789d5a34c256f9/python/kserve/kserve/errors.py#L98